src/operators.jl

```diff
@@ -93,7 +90,14 @@ See [`reverse`](@ref) for a non-modifying version.
 """
 function reverse! end
 @traitfn function reverse!(g::G::IsDirected) where G<:AbstractSimpleGraph
-    g.fadjlist, g.badjlist = g.badjlist, g.fadjlist
+    fadjlist = SimpleGraphs.fadj(g)
+    badjlist = SimpleGraphs.badj(g)
+    @inbounds for i in vertices(g)
+        f_i = copy(fadjlist[i])
+        b_i = copy(badjlist[i])
+        fadjlist[i] = b_i
+        badjlist[i] = f_i
```

Comment:
```suggestion
        fadjlist[i], badjlist[i] = badjlist[i], fadjlist[i]
```

This is a pretty big performance regression though. Before `reverse!` was substantially instantaneous, whereas now it's `O(V)`. I imagine one should at least provide a fast non-mutating aliasing version.

Reply:
Thanks for the comment. This was the one method I was worried about and I made the suggested change. One can always use `reverse` instead but I'm not sure what more can be done here.

Reply:
`reverse` does a copy of the adjacency matrices as far as I understand, to avoid returning aliased graphs, so that would also not fix the performance problem.